apps/emqx_connector/src/emqx_connector_mysql.erl

```diff
@@ -305,6 +315,8 @@ prepare_sql_to_conn(Conn, [{Key, SQL} | PrepareList]) when is_pid(Conn) ->
             ?SLOG(info, LogMeta#{result => success}),
             prepare_sql_to_conn(Conn, PrepareList);
         {error, Reason} ->
+            % FIXME: we should try to differ on transient failers and
```

Comment:
Possible to fix now?

Reply:
I took a shot at it but came to the conclusion that it might require a more thorough refactor. The code is hard to follow as-is and already has several layers of error coping mechanisms, and these should be unified. Created EMQX-8582 to track it.